music21/stream/__init__.py

```diff
@@ -2488,12 +2470,10 @@ def insertAndShift(self, offsetOrItemOrList, itemOrNone=None):
                     #                         'o+shiftPos', o+shiftPos])
 
                     # need original offset, shiftDur, plus the distance from the start
-                    self.setElementOffset(e, o + shiftPos)
+                    self.coreSetElementOffset(e, o + shiftPos)
```

Comment:
outside of the for loop, this should have a coreElementsChanged(), before the insert() statement below, because insert() assumes (or is allowed to assume) that the stream is not dirty.